### PR TITLE
transferred JS code in "archive.html" into seperate file "archive.js".

### LINKED ISSUE #1182: Transfer js code in archive.html into separate .js file
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from underrepresented groups in free and open-source software!

We know that the process of creating a pull request is one of the biggest barriers for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Claim the issue by commenting. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

<details>
  <summary>💡 <i>Learn how to claim</i> 🙋</summary>
  <h3>Claiming an issue</h3>
  <p>Unless the issue is marked as reserved for someone, you can just say "I'd like to try this!" and then you've claimed it - no need to wait for someone to assign it to you. Just be sure you link your pull request (PR) to this issue so we can see where your solution is.</p>
  <p>And open one early if possible - even before you've completed it with additional commits - and others can help you figure out any issues you may face.</p>
</details>

- [ ] 📝 **Update** the file [archive.html](https://github.com/publiclab/Leaflet.DistortableImage/blob/main/examples/archive.html) in the `https://github.com/publiclab/Leaflet.DistortableImage` repository (press the little pen Icon) and edit the line as shown below.

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:
In file `archive.html` insert the code `<script src="./js/archive.js" defer></script>` in the header section, then apply the diff below.
```diff
- <script>
-   let map;
-   let welcomeModal = document.getElementById('welcomeModal');
-   let sidebar = document.getElementById('offcanvasRight');
-   let form = document.getElementById('form');
-   let input = document.getElementById('input');
-   let responseText = document.getElementById('response');
-   let imageContainer = document.getElementById('imgContainer');
-
-   const setupMap = () => {
-     map = L.map('map').setView([51.505, -0.09], 13);
-
-     map.attributionControl.setPosition('bottomleft');
-
-     map.addGoogleMutant();
-
-     map.whenReady(() => {
-       new bootstrap.Modal(welcomeModal).show();
-     });
-   };
-
-   const setupCollection = () => {
-     map.imgGroup = L.distortableCollection().addTo(map);
-   };
-
-   setupMap();
-   setupCollection();
-
-   form.addEventListener('submit', (event) => {
-     event.preventDefault();
-     const url = input.value.replace('details', 'metadata');
-     let fetchedFrom
-     axios.get(url)
-        .then((response) => {
-         if (response.data.files && response.data.files.length != 0) {
-           let imageCount = 0;
-           response.data.files.forEach(file => {
-             if (file.format === 'PNG' || file.format === 'JPEG'){
-               let imageRow = document.createElement('div');
-               let image = new Image(150, 150);
-               let placeButton = document.createElement('a');
-               fetchedFrom = document.createElement('p');
-               fetchedFromUrl = document.createElement('a');
-               fetchedFromUrl.setAttribute('href', input.value);
-               fetchedFromUrl.setAttribute('target', '_blank');
-               fetchedFromUrl.innerHTML = 'this Internet Archive Collection';
-               fetchedFrom.appendChild(fetchedFromUrl)
-
-               placeButton.classList.add('btn', 'btn-sm', 'btn-outline-secondary', 'place-button');
-               placeButton.innerHTML = 'Place on map ';
-
-               placeButton.addEventListener('click', (event) => {
-                 if(!placeButton.getElementsByClassName('fa-spinner')[0]){
-                   const spinner = document.createElement('i')
-                   spinner.setAttribute('class', "fa fa-spinner fa-spin")
-                   spinner.setAttribute('font-family', 'font awesome')
-
-                   placeButton.appendChild(spinner)
-
-                   let imageURL = event.target.previousElementSibling.src;
-                   let image = L.distortableImageOverlay(imageURL);
-                   map.imgGroup.addLayer(image)
-
-                   image.addEventListener('load', () => {
-                     placeButton.removeChild(spinner)
-                   })
-                 }
-               })
-
-               image.src = `${url.replace('metadata', 'download')}/${file.name}`;
-
-               imageRow.classList.add('d-flex', 'justify-content-between', 'align-items-center', 'mb-4', 'pe-5');
-               imageRow.append(image, placeButton);
-
-               imageContainer.appendChild(imageRow);
-               imageCount++;
-             }
-           });
-           responseText.innerHTML = imageCount ? `${imageCount} image(s) fetched successfully from ${fetchedFrom.innerHTML}.` : 'No images found in the link provided...'
-         } else {
-           responseText.innerHTML = 'No images found in the link provided...'
-         }
-       })
-       .catch((error) => {
-          responseText.innerHTML = 'Uh-oh! Something\'s not right with the link provided!'
-       })
-       .finally(() => {
-         bootstrap.Modal.getInstance(welcomeModal).hide();
-       });
-   });
-
-   welcomeModal.addEventListener('hidden.bs.modal', (event) => {
-     new bootstrap.Offcanvas(sidebar).show();
-   });
-
- </script>
```
Create file `archive.js` in folder `/examples/js`, then apply the diff below. That is, you will simply copy the code below and past same in the newly created file.
```diff
+     let map;
+     let welcomeModal = document.getElementById('welcomeModal');
+     let sidebar = document.getElementById('offcanvasRight');
+     let form = document.getElementById('form');
+     let input = document.getElementById('input');
+     let responseText = document.getElementById('response');
+     let imageContainer = document.getElementById('imgContainer');
+
+     const setupMap = () => {
+       map = L.map('map').setView([51.505, -0.09], 13);
+
+       map.attributionControl.setPosition('bottomleft');
+
+       map.addGoogleMutant();
+
+       map.whenReady(() => {
+         new bootstrap.Modal(welcomeModal).show();
+       });
+     };
+
+     const setupCollection = () => {
+       map.imgGroup = L.distortableCollection().addTo(map);
+     };
+
+     setupMap();
+     setupCollection();
+
+     form.addEventListener('submit', (event) => {
+       event.preventDefault();
+       const url = input.value.replace('details', 'metadata');
+       let fetchedFrom
+       axios.get(url)
+          .then((response) => {
+           if (response.data.files && response.data.files.length != 0) {
+             let imageCount = 0;
+             response.data.files.forEach(file => {
+               if (file.format === 'PNG' || file.format === 'JPEG'){
+                 let imageRow = document.createElement('div');
+                 let image = new Image(150, 150);
+                 let placeButton = document.createElement('a');
+                 fetchedFrom = document.createElement('p');
+                 fetchedFromUrl = document.createElement('a');
+                 fetchedFromUrl.setAttribute('href', input.value);
+                 fetchedFromUrl.setAttribute('target', '_blank');
+                 fetchedFromUrl.innerHTML = 'this Internet Archive Collection';
+                 fetchedFrom.appendChild(fetchedFromUrl)
+
+                 placeButton.classList.add('btn', 'btn-sm', 'btn-outline-secondary', 'place-button');
+                 placeButton.innerHTML = 'Place on map ';
+
+                 placeButton.addEventListener('click', (event) => {
+                   if(!placeButton.getElementsByClassName('fa-spinner')[0]){
+                     const spinner = document.createElement('i')
+                     spinner.setAttribute('class', "fa fa-spinner fa-spin")
+                     spinner.setAttribute('font-family', 'font awesome')
+
+                     placeButton.appendChild(spinner)
+
+                     let imageURL = event.target.previousElementSibling.src;
+                     let image = L.distortableImageOverlay(imageURL);
+                     map.imgGroup.addLayer(image)
+
+                     image.addEventListener('load', () => {
+                       placeButton.removeChild(spinner)
+                     })
+                   }
+                 })
+
+                 image.src = `${url.replace('metadata', 'download')}/${file.name}`;
+
+                 imageRow.classList.add('d-flex', 'justify-content-between', 'align-items-center', 'mb-4', 'pe-5');
+                 imageRow.append(image, placeButton);
+
+                 imageContainer.appendChild(imageRow);
+                 imageCount++;
+               }
+             });
+             responseText.innerHTML = imageCount ? `${imageCount} image(s) fetched successfully from ${fetchedFrom.innerHTML}.` : 'No images found in the link provided...'
+           } else {
+             responseText.innerHTML = 'No images found in the link provided...'
+           }
+         })
+         .catch((error) => {
+            responseText.innerHTML = 'Uh-oh! Something\'s not right with the link provided!'
+         })
+         .finally(() => {
+           bootstrap.Modal.getInstance(welcomeModal).hide();
+         });
+     });
+
+     welcomeModal.addEventListener('hidden.bs.modal', (event) => {
+       new bootstrap.Offcanvas(sidebar).show();
+     });
+
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### 🤔❓ Questions?

Leave a comment below!

### Is someone else already working on this?

We encourage you to link to this issue by mentioning the issue # in your pull request, so we can see if someone's already started on it. **If someone seem stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://publiclab.github.io/community-toolbox/#r=all). Thanks!

(This issue was created by [First-Timers-Bot](https://github.com/hoodiehq/first-timers-bot).)

### PR DESCRIPTION
**Pull request by Aarav238**
Fixes #1182  

transferred JS code in `archive.html` into seperate file `archive.js` and added script source in head section of archive.html.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updates
* [ ] @mention the original creator of the issue in a comment below for help or for a review

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by Aarav238** [comment]
@TildaDares  Please review it.

**Comment by TildaDares** [comment]
Hi @Aarav238, contributors can only work on one `first-timers-only` issue. Do you think you could close one of these PRs #1282 or #1285 so that other first-timers have something to work on? Thank you!

**Comment by Aarav238** [comment]
I have raise two pull request of FTO issues.
None of them are merged

**Comment by Aarav238** [comment]
Okay, i have closed the #1285 one.
Please review this one😇.
@TildaDares

**Comment by Aarav238** [comment]
@TildaDares is there any problem in this PR?

**Review by TildaDares — CHANGES_REQUESTED** [review]

**Comment by 7malikk** [comment]
Hello @Aarav238, there have been some new changes to the archive.html file. 
Would you be willing to update your PR accordingly?

**Comment by Aarav238** [comment]
@7malikk i have resolved thr conflict

**Comment by 7malikk** [comment]
@Aarav238 This is looking really good, one final touch, and I believe we are on our way to merge-ville.
Now as your PR implies, extract the js code into a separate file, like you did previously and we are home free.

How does that sound @Aarav238 ?

**Comment by Aarav238** [comment]
but I have Already extracted js code into archieve.js
@7malikk

**Comment by Aarav238** [comment]
Oh It's Okay @7malikk.
Thank you very much.

**Comment by Aarav238** [comment]
@7malikk  I have some doubts related to start contributing in core project in this organisation

**Comment by 7malikk** [comment]
> @7malikk  I have some doubts related to start contributing in core project in this organisation
> 
> 

Oh alright @Aarav238 
You can join the community element chat, and raise said doubts and we can thrash them out there. 

How does that sound?

**Comment by 7malikk** [comment]
@Aarav238 you could start here https://code.publiclab.org/#r=all

**Comment by 7malikk** [comment]
@Aarav238 you could start here https://code.publiclab.org/#r=all

File: examples/js/archive.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+let map;
```
Comment (jywarren) [inline examples/js/archive.js:1]: Hi all, this looks good, but @7malikk would you be able to check this file against the most recent changes in https://github.com/publiclab/Leaflet.DistortableImage/pull/1286/files, or have you already? I just want to be sure we don't lose that last change regarding the search box. Thanks!

Comment (7malikk) [reply examples/js/archive.js:1]: Just went through it again, seems @Aarav238 might've missed the changes regarding the search box.

@Aarav238 would you be willing to fetch the most recent changes files? 

Thank you @jywarren for that pointer.

**Comment by Aarav238** [comment]
When will my PR get merged @jywarren

**Comment by 7malikk** [comment]
@Aarav238 Hey there, could you check your extracted js code again as it is not in line with the code found here: https://github.com/publiclab/Leaflet.DistortableImage/blob/main/examples/archive.html

You could use the command `git fetch upstream/main` to get the latest updates locally and make your own updates.

**Comment by Aarav238** [comment]
Okay @7malikk  Let me try . 
If I will face any problem I will tag you in gitter.😊

**Comment by Aarav238** [comment]
What i have to add in archieve.js @7malikk

**Comment by 7malikk** [comment]
> What i have to add in archieve.js @7malikk 

Hello @Aarav238 
All you need to do is to extract the recently updated js code in the archive.html

https://github.com/publiclab/Leaflet.DistortableImage/blob/main/examples/archive.html

File: examples/archive.html [hunk 2]
```diff
@@ -17,6 +17,19 @@
...
+  <body style="margin:0;">
+    
+    <div class="modal fade" id="welcomeModal" aria-labelledby="exampleModalLabel" tabindex="-1" aria-hidden="true">
+      <div class="modal-dialog modal-dialog-centered modal-dialog-scrollable">
+        <div class="modal-content bg-light">
+          <div class="modal-header d-block">
+            <div class="d-flex justify-content-between align-items-center">
+              <h2 class="modal-title">Welcome to MapKnitter Lite</h2>
+              <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close" ></button>
+
```
Comment (7malikk) [inline examples/archive.html:32]: Hey @Aarav238, great work so far, this snippet seems to be in the wrong place, could you go over it again? Thanks!

Comment (Aarav238) [reply examples/archive.html:32]: I haven't understood what is the issue here @7malikk . Please elaborate little bit more.

Comment (7malikk) [reply examples/archive.html:32]: Hey there @Aarav238 If you take a look at this snippet of your file

https://github.com/publiclab/Leaflet.DistortableImage/blob/b9511d3533fac214a97fe5caee1f3d58bc5e8933/examples/archive.html#L21-L44

You'd notice that your file has two `</head>` tags at lines 21 and 35 respectively 
You'd also notice there are two `<body>` tags at lines 23 and 36 respectively

This further implies that there are duplicate codes within your file.

Would you be willing to make a correction?

Comment (Aarav238) [reply examples/archive.html:32]: Yeah sure.

**Comment by Aarav238** [comment]
I have resolved the issue.

**Comment by 7malikk** [comment]
@Aarav238 Hey there, great work so far
One last thing, here:
https://github.com/publiclab/Leaflet.DistortableImage/blob/073e121f1ebc7e5f326c5b402d89fe61258308d4/examples/archive.html#L75-L201

I believe you forgot to cut the js code from the html file.
Would you like to resolve this? This should be the last of them all.

**Comment by jywarren** [comment]
Great work, folks!!!

**Pull request merged**